src/dodal/common/watcher_utils.py

```diff
@@ -0,0 +1,80 @@
+from ophyd_async.core import WatchableAsyncStatus, Watcher
+
+from dodal.log import LOGGER
+
+Number = int | float
+
+
+class _LogOnPercentageProgressWatcher(Watcher[Number]):
+    def __init__(
+        self,
+        status: WatchableAsyncStatus[Number],
+        message_prefix: str,
+        percent_interval: Number = 25,
+    ):
+        status.watch(self)
+        self.percent_interval = percent_interval
+        self._current_percent_interval = 0
+        self.message_prefix = message_prefix
+        if self.percent_interval <= 0:
+            raise ValueError(
+                f"Percent interval on class _LogOnPercentageProgressWatcher must be a positive number, but recieved {self.percent_interval}"
+            )
+
+    def __call__(
+        self,
+        current: Number | None = None,
+        initial: Number | None = None,
+        target: Number | None = None,
+        name: str | None = None,
+        unit: str | None = None,
+        precision: int | None = None,
+        fraction: float | None = None,
+        time_elapsed: float | None = None,
+        time_remaining: float | None = None,
+    ):
+        if (
+            isinstance(current, Number)
+            and isinstance(target, Number)
+            and isinstance(initial, Number)
+            and target - initial
```

Comment:
Could: I guess this is to check that `target != initial`? It feels a bit unintuitive, feels like just having `target != initial` here is clearer?